### ISSUE TITLE
Resetting max_radii2D in densification_postfix() seems to make no gaussians pruned in densify_and_prune(). 

### ISSUE THREAD

**Issue by StarSapph1re** [graphdeco-inria/gaussian-splatting#741]
In function

```
    def densify_and_prune(self, max_grad, min_opacity, extent, max_screen_size):
        grads = self.xyz_gradient_accum / self.denom
        grads[grads.isnan()] = 0.0
        self.densify_and_clone(grads, max_grad, extent)
        self.densify_and_split(grads, max_grad, extent)
        prune_mask = (self._opacity < min_opacity).squeeze()
        print("opacity true =", torch.sum(prune_mask).item())
        if max_screen_size:
            big_points_vs = self.max_radii2D > max_screen_size
            prune_mask = torch.logical_or(prune_mask, big_points_vs)
            print("radii true = ", torch.sum(prune_mask).item())
            big_points_ws = self.get_scaling.max(dim=1).values > 50 * extent
            prune_mask = torch.logical_or(prune_mask, big_points_ws)
            print("scaling true = ", torch.sum(prune_mask).item())
        self.prune_points(prune_mask)
        torch.cuda.empty_cache()
```

densify_and_split() will be called first, but this function will call densification_postfix():

```
    def densification_postfix(self, new_xyz, new_features_dc, new_features_rest, new_opacities, new_scaling,
                              new_rotation, new_transform):
        d = {"xyz": new_xyz,
             "f_dc": new_features_dc,
             "f_rest": new_features_rest,
             "opacity": new_opacities,
             "scaling": new_scaling,
             "rotation": new_rotation,
             "tau": new_transform}

        optimizable_tensors, optimizable_tensors_t = self.cat_tensors_to_optimizer(d)
        self._xyz = optimizable_tensors["xyz"]

        self._features_dc = torch.cat((self._features_dc, new_features_dc), dim=0)
        self._features_rest = torch.cat((self._features_rest, new_features_rest), dim=0)
        self._opacity = torch.cat((self._opacity, new_opacities), dim=0)
        # print("new opa=", new_opacities.shape)

        self.tilted.tilted.tau = optimizable_tensors_t["tau"]
        self._scaling = optimizable_tensors["scaling"]
        self._rotation = optimizable_tensors["rotation"]

        self.xyz_gradient_accum = torch.zeros((self.get_xyz.shape[0], 1), device="cuda")
        self.denom = torch.zeros((self.get_xyz.shape[0], 1), device="cuda")

        self.max_radii2D = torch.zeros((self.get_xyz.shape[0]), device="cuda")
        # Here it just set self.max_radii2D to all zeros
```
densification_postfix() will set self.max_radii2D to all zeros, will this result in no gaussians will be pruned for being too large in the value max_radii2D ? When I ran these other's code in other's project, I printed the number of True values in the mask for pruning gaussians with max_radii2D value above the threshold and got always 0 along the whole training progress. I wonder if I made some mistakes or misunderstand the pruning progress.

Looking forward to your reply! Thanks so much :)

**Comment by LarkLeeOnePiece** [comment]
You should check this issue :https://github.com/graphdeco-inria/gaussian-splatting/issues/123

**Comment by StarSapph1re** [comment]
Thanks！I didn't find it when I searched the keyword max_radii2D.
